src/openrct2/world/tile_inspector.c

```diff
@@ -0,0 +1,338 @@
+	*firstElement = *secondElement;
+	*secondElement = temp;
+
+	// Swap the 'last map element for tile' flag if either one of them was last
+	if (map_element_is_last_for_tile(firstElement) || map_element_is_last_for_tile(secondElement))
+	{
+		firstElement->flags ^= MAP_ELEMENT_FLAG_LAST_TILE;
+		secondElement->flags ^= MAP_ELEMENT_FLAG_LAST_TILE;
+	}
+}
+
+/**
+ * Inserts a corrupt element under a given element on a given tile
+ * @param x, y: The coordinates of the tile
+ * @param element_index: The nth element on this tile
+ * Returns 0 on success, MONEY_UNDEFINED otherwise.
+ */
+sint32 tile_inspector_insert_corrupt_at(sint32 x, sint32 y, sint16 element_index, sint32 flags)
+{
+	// Make sure there is enough space for the new element
+	if (!map_check_free_elements_and_reorganise(1))
+	{
+		return MONEY32_UNDEFINED;
+	}
+
+	if (flags & GAME_COMMAND_FLAG_APPLY)
+	{
+		// Create new corrupt element
+		rct_map_element *curruptElement = map_element_insert(x, y, -1, 0); // Ugly hack: -1 guarantees this to be placed first
+		if (curruptElement == NULL)
+		{
+			log_warning("Failed to insert corrupt element.");
+			return MONEY32_UNDEFINED;
+		}
+		curruptElement->type = MAP_ELEMENT_TYPE_CORRUPT;
+
+		// Set the base height to be the same as the selected element
+		rct_map_element *const selectedElement = map_get_first_element_at(x, y) + element_index;
+		curruptElement->base_height = curruptElement->clearance_height = selectedElement->base_height;
+
+		// Move the corrupt element up until the selected list item is reached
+		// this way it's placed under the selected element, even when there are multiple elements with the same base height
+		for (sint16 i = 0; i < element_index; i++)
+		{
+			map_swap_elements_at(x, y, i, i + 1);
+		}
+
+		map_invalidate_tile_full(x << 5, y << 5);
+
+		// Update the tile inspector's list for everyone who has the tile selected
+		rct_window *const tile_inspector_window = window_find_by_class(WC_TILE_INSPECTOR);
+		if (tile_inspector_window != NULL && (uint32)x == windowTileInspectorTileX && (uint32)y == windowTileInspectorTileY)
+		{
+			windowTileInspectorElementCount++;
+
+			// Keep other elements (that are not being hidden) selected
+			if (tile_inspector_window->selected_list_item > element_index)
+			{
+				tile_inspector_window->selected_list_item++;
+			}
+
+			if (tile_inspector_window->selected_list_item == element_index)
+			{
+				window_tile_inspector_set_page(tile_inspector_window, PAGE_CORRUPT);
+			}
+
+			window_tile_inspector_auto_set_buttons(tile_inspector_window);
+			window_invalidate(tile_inspector_window);
+		}
+	}
+
+	// Nothing went wrong
+	return 0;
+}
+
+/**
+* Forcefully removes an element for a given tile
+* @param x, y: The coordinates of the tile
+* @param element_index: The nth element on this tile
+*/
+sint32 tile_inspector_remove_element_at(sint32 x, sint32 y, sint16 element_index, sint32 flags)
+{
+	if (flags & GAME_COMMAND_FLAG_APPLY)
+	{
+		// Forcefully the element
```

Comment:
@Broxzier please don't take any offence. It made me laugh though, I found it funny.